[Darren Heng] iP

darren2pro opened the pull request:
# Naruto
> "I never go back on my word. That is my Ninja Way!" - Naruto [Narutopedia page](https://naruto.fandom.com/wiki/Naruto_Uzumaki)

Naruto will help you to manage your time better by keeping track of your:
* Deadlines
* Events
* ToDos

It is ~~FAST~~ SUPER FAST to use too!

All you need to do is,
1. Download it from [here](https://github.com/darren2pro/ip/releases/tag/v0.1)
2. Click on Duke.java.jar
3. Run the jar file on your local computer
4. And we're done!

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Whacky cool GUI
- [ ] Reminders (coming soon :grin:)

If you are a Java programmer, you can use it to practice Java too! Here's the `main` method for the `Launcher` class.

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

darren2pro committed:
Complete Level-1

darren2pro committed:
Complete Level-2

darren2pro committed:
Complete Level-3

darren2pro committed:
Complete Level-4

darren2pro committed:
Complete A-TextUiTesting

darren2pro committed:
Complete A-TextUiTesting part 2

darren2pro committed:
Complete Level-5

darren2pro committed:
Complete Level-6

darren2pro committed:
Complete Level-7

darren2pro committed:
Complete Level-8

darren2pro committed:
Merge branch 'branch-Level-7'

darren2pro committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadlines.java

darren2pro committed:
Improve Level-8 to include java.time.format.DateTimeFormatter.ISO_DATE

darren2pro committed:
Add Ui, *Command, Storage, Parser, TaskList

darren2pro committed:
Complete A-MoreOOP

darren2pro committed:
Complete A-Packages

darren2pro committed:
Write JUnit test cases part 1

darren2pro committed:
Complete A-JUnit

darren2pro committed:
Complete A-Jar

darren2pro committed:
Complete A-JavaDoc

darren2pro committed:
Complete A-CodingStandard

darren2pro committed:
Further tweaks to the JavaDocs

darren2pro committed:
Complete Level-9

darren2pro committed:
Merge branch 'branch-A-JavaDoc'

darren2pro committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/commands/Command.java
#	src/main/java/duke/tasks/Deadlines.java
#	src/main/java/duke/tasks/Events.java
#	src/main/java/duke/utils/TaskList.java
#	src/main/java/duke/utils/Ui.java

darren2pro committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/commands/Command.java
#	src/main/java/duke/utils/Ui.java

darren2pro committed:
Change constant variable names to screaming snake case

darren2pro committed:
Merge remote-tracking branch 'origin/add-gradle-support'

darren2pro committed:
Merge branch 'master' into A-Gradle

darren2pro committed:
Automate project builds using Gradle

darren2pro committed:
Merge branch 'A-Gradle'

darren2pro committed:
Add checkstyle.xml and suppressions.xml

darren2pro committed:
Merge branch 'master' into A-CheckStyle

darren2pro committed:
Rectify checkstyle violations

darren2pro committed:
Merge branch 'A-CheckStyle'

darren2pro committed:
Merge branch 'master' into branch-Level-10

darren2pro committed:
Add launcher class

darren2pro committed:
Add some GUI capabilities without linking it to Duke

Ruofan1023 reviewed (commented):
You code is pretty neat and follows coding standard in general. Great work!

darren2pro committed:
Link GUI to duke to a certain extent

darren2pro commented:
> You code is pretty neat and follows coding standard in general. Great work!

Thank you!

darren2pro committed:
Minor tweaks to improve GUI

darren2pro committed:
Merge branch 'branch-Level-10'

mukundrs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,112 @@
+
+        // Now execute the entire command from user. It goes to my special stream
+        try {
+            Command c = Parser.parse(fullCommand);
```
A better name could be used for command instead of c. Could just use command instead

mukundrs commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,254 @@
+            super.setExitTrue();
+        }
+    }
+
```
Extra empty line here

mukundrs reviewed (commented):
Your code looks really good to me! Very minor nitpicks here and there but apart from that excellent work!

Ruofan1023 commented on src/main/java/duke/Duke.java:
```diff
@@ -46,7 +60,53 @@ public void run() {
+        ByteArrayOutputStream narutoStream = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(narutoStream);
+        // IMPORTANT: Save the old System.out!
+        PrintStream old = System.out;
```
a better name might be "oldPrintStream".

darren2pro replied:
That's true!

Ruofan1023 commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+}
```
Empty line at end of the file (LoL)

Ruofan1023 reviewed (commented):
Good work and couldn't find much to criticise!

darren2pro committed:
Update build.gradle for the new JAR file

darren2pro committed:
Add a few important assertions into the code base

We make some assumptions in our code that the task number we give
to `TaskList::getTask` and `TaskList::markTaskAsDone` is within
valid range. We also make some assumptions that the task we want
to mark as done is not null.

This could cause unexpected errors when the tasks we retrieve
ends up being null, or that the task number that we pass to the
function ends up being out of range.

Let's
1. Assert that the task we want to get is not null
2. Assert that the task we want to mark as done is not null
3. Assert that the task number supplied to the
and  is within valid range.

darren2pro committed:
Improve code quality

There are some occurrences of magic string literals that appear
in the code. In addition, these string literals appear in numerous
different places.

This would be troublesome when we want to edit
these string literals because we would have to edit multiple
methods. Some examples are the string prefix that we add to all of
Naruto's reply, and the error message given when we encounter an
error while accessing persisted data.

Also, there are multiple unused methods, attributes and
constructors, e.g. the Duke class's default constructor.

These cause unnecessary clutter in the code and make it more
difficult to understand.

Let's
* Save the magic string literals as a final attribute within the
class.
* Remove unused methods, attributes and constructors.

darren2pro committed:
Include the use of Java Streams in the code (already included)

darren2pro committed:
Merge pull request #1 from darren2pro/branch-A-Assertions

Merge A-Assertions using PR

darren2pro committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

darren2pro committed:
Merge pull request #2 from darren2pro/branch-A-CodeQuality

Merge A-CodeQuality using PR

darren2pro committed:
Merge branch 'master' into branch-A-Streams

darren2pro committed:
Merge pull request #3 from darren2pro/branch-A-Streams

Merge A-Streams using PR

darren2pro committed:
Complete C-Sort

darren2pro committed:
Merge branch 'branch-BCD-Extension'

darren2pro committed:
Refactor - Extract static nested classes into their own class

darren2pro committed:
Improve code quality

darren2pro committed:
Update README.md

darren2pro committed:
Set theme jekyll-theme-slate

darren2pro committed:
Tweak docs/README.md to correct formatting errors

darren2pro committed:
Merge branch 'master' of https://github.com/darren2pro/ip

darren2pro committed:
Add capability to terminate program upon "bye" command

darren2pro committed:
Remove GitHub emoji from docs/README.md

darren2pro committed:
Correct formatting errors

darren2pro committed:
Correct checkstyle violations
